config/admin/headers.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+/** Content-Security-Policy headers */
+header("Content-Security-Policy-Report-Only: "
+    ."default-src 'self'; "
+    ."style-src 'self' 'unsafe-inline' https://fonts.googleapis.com https://cdnjs.cloudflare.com https://*.bootstrapcdn.com; "
+    ."script-src 'self' 'unsafe-inline' 'unsafe-eval' https://cdnjs.cloudflare.com; "
+    ."img-src 'self' data: https://cdnjs.cloudflare.com; "
+    ."font-src 'self' data: https://*.bootstrapcdn.com https://fonts.gstatic.com; "
+    ."form-action 'self'; "
+    ."upgrade-insecure-requests; "
+    ."block-all-mixed-content; "
+    ."report-uri /csp.php");
```

Comment:
I think listing the directives in an array and imploding that is more readable and easily extendable:

```suggestion
<?php

/**
 * Adminland: HTTP Content-Security-Policy (CSP) Headers
 */

$directives = [
    "default-src 'self'",
    "font-src 'self' data: https://*.bootstrapcdn.com https://fonts.gstatic.com",
    "form-action 'self'",
    "img-src 'self' data: https://cdnjs.cloudflare.com",
    "script-src 'self' 'unsafe-inline' 'unsafe-eval' https://cdnjs.cloudflare.com",
    "style-src 'self' 'unsafe-inline' https://fonts.googleapis.com https://cdnjs.cloudflare.com https://*.bootstrapcdn.com",
    'block-all-mixed-content',
    'upgrade-insecure-requests',
    'report-uri /csp.php',
];

header('Content-Security-Policy-Report-Only: '.implode('; ', $directives));
```

Reply:
The `/csp.php` file is missing.

Reply:
I think these HTTP headers should be defined in a service provider (maybe the `AdminServiceProvider`) to tie into the initial PSR-7 Response object created by Slim.